mobie/metadata/slice_grid_view.py

```diff
@@ -0,0 +1,197 @@
+        raise ValueError("create_slice_grid is only supported for image sources.")
+
+    z_axis = 0
+    shape = _load_shape(source_data["imageData"])
+    ax_len = shape[z_axis]
+    # spacing_pixels = shape[z_axis] / n_slices
+    if ax_len % n_slices != 0:
+        msg = "Can't evenly split volume with length {ax_len} into {n_slices} slices.\n"
+        msg += "The space between slices will be {spacing_pixels} pixels."
+        warnings.warn(msg)
+
+    # load the data transformation and compose it with the
+    data_transform = _load_transformation(source_data["imageData"])
+    if initial_transforms is not None:
+        pass  # TODO
+
+    # compute the individual slice transformations (shifts)
+    # to enable axis != 0 we would also need to do an axis rotation here
+    source_transformations = []
+    grid_sources = []
+    for slice_id in range(n_slices):
+        # TODO need to bring 'data_transform' and 'shape' into correct shapes and types so that this works;
+        # will need some functionaliy from elf.transformation for this.
+        # and need to find correct shift for this slice
+        slice_trafo_params = data_transform @ shape
+        name_after_trafo = f"{source}_z{slice_id}"
+        slice_trafo = view_utils.get_affine_source_transform(
+            [source], slice_trafo_params, source_names_after_transform=[name_after_trafo]
+        )
+        source_transformations.append(slice_trafo)
+        grid_sources.append(name_after_trafo)
+
+    # add the grid transformation
+    grid = view_utils.get_transformed_grid_source_transform(grid_sources)
+    source_transformations.append(grid)
+
+    # TODO
+    if display_settings is None:
+        # if no display settings were passed, use them from the default view for this source
+        if source in views:
+            pass
+        # and if we don't have a default view, use the default imageSource settings
+        else:
+            pass
+
+    new_view = view_utils.get_view([display_name], [source_type], [grid_sources], [display_settings],
+                                   is_exclusive=is_exclusive, menu_name=menu_name,
+                                   source_transforms=source_transformations)
+    return new_view
+
+
+# TODO should be unified with functionality from 'bookmark_metadata'
+# (which is a bad name for this file, eventaully need to restrucure this for a 0.5.0 release)
+def _write_view(dataset_folder, dataset_metadata, view_file, view_name, new_view, overwrite):
+    # TODO write to view_file instead if it is not None
+    if view_name in dataset_metadata["views"]:
+        msg = "The view {name} is alread present in the dataset at {dataset_folder}."
+        if overwrite:
+            warnings.warn(msg + " It will be over-written.")
+        else:
+            raise ValueError(msg)
+    dataset_metadata["views"][view_name] = new_view
+    write_dataset_metadata(dataset_folder, dataset_metadata)
+
+
+def create_slice_grid(
```

Comment:
This is the function for creating a slice grid **without** passing an initial reference view.